server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -220,8 +227,8 @@ private void allocateUnassigned() {
                             final var decision = allocation.deciders().canAllocate(shard, routingNode, allocation);
                             switch (decision.type()) {
                                 case YES -> {
-                                    if (logger.isTraceEnabled()) {
-                                        logger.trace("Assigned shard [{}] to [{}]", shard, desiredNodeId);
+                                    if (logger.isDebugEnabled()) {
```

Comment:
nit: no real need for these checks, they're the first thing that `logger.debug()` does anyway